### PR TITLE
Fix clipping and device size bugs

### LINKED ISSUE #47: ggplot2: One line of panel.background always gets dropped
```
library(ggplot2)
library(svglite)

svglite("testplot.svg")
qplot(mtcars$cyl) + theme_bw()
dev.off()
```

testplot

### PR DESCRIPTION
**Pull request by yixuan**
This PR fixes #47 and #50 .

### CODE REVIEW

**Comment by hadley** [comment]
Could you add a unit test for the clipping too please? Maybe something using the problem in #47? (but re-created with base graphics)

**Comment by yixuan** [comment]
Yeah, I've thought about it, but I actually don't know what we really want to test. It's hard to examine the effectiveness of the clipping purely from the generated SVG code. :-|

**Comment by hadley** [comment]
Another way to attack this is a regression test: we just save the svg output in the tests directory, so we can at least see if things change.  That's particularly nice in combination with github's image diffs, but unfortunately you have to push to see those.

**Comment by yixuan** [comment]
I agree. I created a "fake" test that will create a plot to highlight the effect of clipping. We can actually see the difference as the plot below shows:

test-clip

**Comment by hadley** [comment]
Can you please check that svg file in too?

File: src/devSVG.cpp [hunk 1]
```diff
@@ -418,6 +457,13 @@ void svg_text(double x, double y, const char *str, double rot,
               double hadj, const pGEcontext gc, pDevDesc dd) {
   SVGDesc *svgd = (SVGDesc*) dd->deviceSpecific;
 
+  // If we specify the clip path inside <text>, the "transform" also
+  // affects the clip path, so we need to specify clip path at an outer level
+  if(svgd->clipno > 0) {
```
Comment (hadley) [inline src/devSVG.cpp]: All these `if` need a space (`if (`) to be consistent with the rest of the style

Comment (yixuan) [inline src/devSVG.cpp]: Good catch.

**Comment by yixuan** [comment]
Do you mean adding and committing the `test-clip.svg` file to the test folder?

**Comment by hadley** [comment]
@yixuan yup

**Comment by hadley** [comment]
Can you double check that clipPath needs to go in defs? I think that's only necessary if you want to embed at the top of the document, but I might be wrong.

**Comment by yixuan** [comment]
It is not necessary, but recommended. I quote the following paragraph from the [SVG standard](https://www.w3.org/TR/SVG/struct.html#Head):

> It is recommended that, wherever possible, referenced elements be defined inside of a ‘defs’ element. Among the elements that are always referenced: ‘altGlyphDef’, ‘clipPath’, ‘cursor’, ‘filter’, ‘linearGradient’, ‘marker’, ‘mask’, ‘pattern’, ‘radialGradient’ and ‘symbol’. Defining these elements inside of a ‘defs’ element promotes understandability of the SVG content and thus promotes accessibility.

**Comment by hadley** [comment]
Ok, great - thanks!

**Comment by hadley** [comment]
For future reference, when a pull request closes issues, can you please include in the commit messages so that they get closed automatically on merge? (i.e. include `Fixes #47` in one of the commits)

**Comment by hadley** [comment]
Oh I guess you did, but missed `#50`

**Comment by yixuan** [comment]
Sorry I didn't. :-\
It seems that I'm unaware of all these fancy tricks on Github.

**Pull request merged**